spyder/plugins/editor/plugin.py

```diff
@@ -2284,6 +2273,85 @@ def new(self, fname=None, editorstack=None, text=None):
             if index is not None and not current_es.close_file(index):
                 return
 
+        # If no text is provided, create default content
+        try:
+            if text is None:
+                default_content = True
+                text, enc = encoding.read(self.TEMPLATE_PATH)
+                enc_match = re.search(r'-*- coding: ?([a-z0-9A-Z\-]*) -*-',
+                                      text)
+                if enc_match:
+                    enc = enc_match.group(1)
+
+                # Initialize template variables
+                if os.name == 'nt':
+                    # Windows
+                    import ctypes
+
+                    username = encoding.to_unicode_from_fs(
+                                os.environ.get('USERNAME', '-'))
+                    GetUserNameEx = ctypes.windll.secur32.GetUserNameExW
+                    name_display = 3
+
+                    size = ctypes.pointer(ctypes.c_ulong(0))
+                    GetUserNameEx(name_display, None, size)
+
+                    name_buffer = ctypes.create_unicode_buffer(
+                                size.contents.value)
+                    GetUserNameEx(name_display, name_buffer, size)
+                    displayname = name_buffer.value.strip()
+                else:
+                    # Linux, Mac OS X
+                    import pwd
+
+                    username = encoding.to_unicode_from_fs(
+                                   os.environ.get('USER', '-'))
+                    try:
+                        pwd_struct = pwd.getpwnam(username)
+                    except KeyError:
+                        displayname = ''
+                    else:
+                        # Use first element in Gecos field as user's full name
+                        displayname = pwd_struct.pw_gecos.split(',')[0].strip()
+
+                now = datetime.now().astimezone()
+                now.replace(microsecond=0)
+                VARS = {
+                    'date': now.ctime(),
+                    'isodate': now.isoformat(),
+                    'year': now.strftime('%Y'),
+                    'month': now.strftime('%m'),
+                    'day': now.strftime('%d'),
+                    'hour': now.strftime('%H'),
+                    'minute': now.strftime('%M'),
+                    'second': now.strftime('%S'),
+                    'tzname': now.strftime('%Z'),
+                    'monthname': now.strftime('%b'),
+                    'weekday': now.strftime('%a'),
+                    'username': username,
+                    'fullname': displayname or 'n/a',
+                    'file': fname,
+                    'filename': os.path.basename(fname),
+                    'filepath': os.path.dirname(fname),
+                    'projectname': project_name or 'n/a',
+                    'projectpath': project_path or 'n/a'
```

Comment:
Not totally sure what to do about these in the case of no project being active. For starters, we should use proper capitalization, and calling `_()` will translate the string, i.e.

```suggestion
                    'projectname': project_name or _('N/A'),
                    'projectpath': project_path or _('N/A'),
```


We _could_ just fall back to `getcwd_or_home()`, as Spyder does for the default working dir when no project is active. i.e.:

```suggestion
                    'projectname': project_name or _('N/A'),
                    'projectpath': project_path or getcwd_or_home(),
```

Reply:
I am used to both ways: ‘n/a’ and ‘N/A’. When working in the UK I got the feeling, that ‘n/a’ was more common. So does www.leo.org only shows ‘n/a’. On the other hand, wikipedia has an entry for ‘N/A’. I don't have any preferences here.

I prefer to show, that the file is not part of a project instead of using the current working directory or even home. Otherwise one might think, that the file is part of a project, which it is not.